packages/dashboard/src/components/palette/icons/index.tsx

```diff
@@ -21,16 +22,46 @@ const PaletteComponentIcon: React.FC<PaletteComponentIconProps> = ({
 }) => {
   const [hover, setHover] = useState<boolean>(false);
 
-  const tooltipStyle = {
-    fontSize: spaceScaledM,
-    color: colorTextButtonNormalActive,
-    backgroundColor: colorBackgroundLayoutMain,
-    padding: `${spaceScaledS} ${spaceScaledM}`,
-    borderRadius: spaceStaticXs,
-    borderWidth: spaceScaledXxxs,
-    borderColor: colorBorderButtonNormalDisabled,
-    boxShadow: `${spaceScaledXxxs} ${spaceScaledXxxs} ${spaceScaledXxxs} ${colorBorderButtonNormalDisabled}`,
-  };
+  const Tooltip = styled.div<{ hover: boolean }>`
```

Comment:
Is there a reason we're not reusing the component exported from `packages/react-components/src/components/tooltip/tooltip.tsx` ?  I dont see why we have to duplicate a similar tooltip that was already done in this PR https://github.com/awslabs/iot-app-kit/pull/2715

Reply:
I made a copy of this branch to try out the approach for reusing `packages/react-components/src/components/tooltip/tooltip.tsx` - https://github.com/awslabs/iot-app-kit/pull/2747

we can reuse this component, but it is not keyboard accessible - on pressing escape key, tooltip won't disappear.
This is existing behavior for this component, when we tried to fix it using a `hover` state and registered an event handler for handling escape, somehow it is lagging in that component.
Another issue is, if you rapidly hover over the widget icons, some of the previously visible tooltips won't disappear.
After spending some more time, we still didn't find the reason behind this.

Can we merge this PR for dark mode / theming support for now? with these changes, it is still accessible(dismiss using esc).

**To improve the tooltip:**
Can we not use some library like react-tooltip which has support for props like `closeOnEsc`?
Or rebuild the tooltip component in such a way that it is accessible?

Reply:
agreed that its better to get this theming support out quickly! thank you for investigating further 🙏 i will add a task to our backlog to consolidate the tooltip implementations!